### PR TITLE
Add syntaxes highlight: if/else, true/false, and/or and others

### LINKED ISSUE #22: True, False, Nothing not highlighted

### PR DESCRIPTION
**Pull request by bopoh13**
Close #22

Part first of two.
- Added 27 items highlighting. For 17 of which autotests are written.  
  Read more in [CHANGELOG.md](https://github.com/serkonda7/vscode-vba/blob/5b66635c00cee973b16129f6442d1eb9c1fddbf9/CHANGELOG.md)
- Changed internal names of code for better readability
- Fixed typos in the repository

In the second part VBA numeric types #10 will be corrected

### CODE REVIEW

**Comment by bopoh13** [comment]
_Any comments are welcome_

**Comment by serkonda7** [comment]
I can give a review in a few hours.

**Comment by bopoh13** [comment]
@serkonda7 please take your time if you get a chance to view. I'm still testing numeric types.

**Review by serkonda7 — CHANGES_REQUESTED** [review]
Nice work overall :rocket: . I think some pattern names require clarification as stated in the comments.
Please let me know what you think about my suggestions.

The tests (after adapting changed names) look good.

**Comment by bopoh13** [comment]
For some reason I couldn't reply under comment.

> Just revert this line

Ok, just the shield looks like button )

> name may be clarified to something like `keyword.procedure.vba`

I wrote `keyword.$1.vba` variant. Will whitespace characters affect autotests?
https://github.com/serkonda7/vscode-vba/blob/15d70fb450432da5c1347243f37029f97a5d37fc/syntaxes/tests/procedure.bas#L4

**Comment by bopoh13** [comment]
**Need revise description:** this PR doesn't solve issue #16
Link: https://docs.microsoft.com/en-us/dotnet/visual-basic/language-reference/statements/class-statement

**Comment by bopoh13** [comment]
> Replaced description of the first commit in PR. Branch _bopoh13:dev-types_ is broken.

**Comment by serkonda7** [comment]
> **Need revise description:** this PR doesn't solve issue #16
> Link: https://docs.microsoft.com/en-us/dotnet/visual-basic/language-reference/statements/class-statement

Sorry, was too fast on this one.

**Comment by serkonda7** [comment]
> I wrote `keyword.$1.vba` variant. Will whitespace characters affect autotests?

I'm not sure about this but will validate.

File: syntaxes/vba.tmLanguage.json [hunk 1]
```diff
@@ -27,63 +26,55 @@
...
-					"name": "keyword.control.End.vba",
-					"match": "\\bEnd (With)\\b"
+					"name": "keyword.$1.vba",
+					"match": "(?i:\\b(Attribute|End( (Function|Property|Sub))?|((Private|Public|Static) )?(Const|Function|Property|Sub))\\b)"
 				}
 			]
 		},
 		"numbers": {
 			"patterns": [
 				{
```
Comment (serkonda7) [inline syntaxes/vba.tmLanguage.json:48]: I can understand you merged the number regexes to shorten them but it was a lot easier to understand their behaviour beforehand.
I'd prefer keeping them separated.

**Review by serkonda7 — CHANGES_REQUESTED** [review]

**Comment by serkonda7** [comment]
Also the test fails:
```
at [syntaxes/tests/procedure.bas:[18](https://github.com/serkonda7/vscode-vba/runs/5827759924?check_suite_focus=true#step:5:18):18:[20](https://github.com/serkonda7/vscode-vba/runs/5827759924?check_suite_focus=true#step:5:20)]:
  18:   Function foo() As Boolean
                       ^^
  missing required scopes: keyword.As.vba
  actual: source.vba keyword.control.vba


  at [syntaxes/tests/procedure.bas:32:20:28]:
  32: Function withParam(Optional p1 As MyType)
                         ^^^^^^^^
  missing required scopes: storage.modifier.visibility
  actual: source.vba keyword.control.vba
```

**Comment by bopoh13** [comment]
> I'd prefer keeping them separated

I missed the idea that there would be a group in the future.

> test fails

Well, I understood my mistake. It was necessary to upgrade from `main` branch.

**Comment by serkonda7** [comment]
> I missed the idea that there would be a group in the future.

I don't see a benefit in the group. It's just less readable. 
Or have I missed a point here? 

> Well, I understood my mistake. It was necessary to upgrade from `main` branch.

I may expand the contribution docs with a guide to resolve merge conflicts.

**Comment by bopoh13** [comment]
> Or have I missed a point here?

I had a little lost where rebase this branch. When installing dev-expansion I didn't see actual files because missed key `--force`.

> I may expand the contribution docs with a guide to resolve merge conflicts.

I understood, judging by [the сommit](
https://github.com/serkonda7/vscode-vba/pull/38/commits/4c55b8d437c5738830413d2ce61d5c299e4d9e76) earlier even squash worked. Thank you

**Review by serkonda7 — APPROVED** [review]
Thanks for the work and discussion. Really nice changes 🎉

**Comment by bopoh13** [comment]
Thanks for the discussion and this project. It may be richer than it seems at first glance.

**Pull request merged**